controllers/subscription_controller.go

```diff
@@ -290,6 +291,22 @@ func (r *SubscriptionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	err := mgr.Add(manager.RunnableFunc(func(ctx context.Context) error {
+
+		odfDepsSub := GetStorageClusterSubscriptions()[0]
+
+		if odfDepsSub.Spec.Package != OdfDepsSubscriptionPackage {
+			panic("odfDepsSub variable is expected to contain the 'odf-dependencies' subscription. " +
+				"Ensure the 'odf-dependencies' subscription indexed at 0.")
```

Comment:
who needs to ensure this? if this is static at build time why not move this to main? may I know what's the requirement? why can't we make an arrangement for caller to explicitly get deps subscription and then the remaining?

Reply:
We developer needs to ensure this while coding. I added a panic so that we can catch this while testing. Maybe I can create a unit test to ensure this.

Reply:
Done